src/components/table/table-row/table-row.style.js

```diff
@@ -0,0 +1,200 @@
+import StyledTableHeader from '../table-header/table-header.style';
+import { isClassic } from '../../../utils/helpers/style-helper';
+import { THEMES } from '../../../style/themes';
+
+const StyledTableRow = css`
+  .carbon-table-row {
+    ${applyRowStyling}
+  }
+`;
+
+function applyRowStyling(props) {
+  const { theme } = props;
+  return css`
+    .custom-drag-layer {
+      & {
+        background-color: #002E41;
+        cursor: grabbing;
+        cursor: -moz-grabbing;
+        cursor: -webkit-grabbing;
+        display: block;
+      }
+
+      ${StyledTableCell} {
+        background-color: #002E41;
+      }
+
+      .configurable-item-row__content-wrapper {
+        visibility: visible;
+      }
+    }
+
+    &:first-child ${StyledTableHeader} {
+      &:first-child {
+        border-radius: 0px 0 0 0;
+      }
+
+      &:last-child {
+        border-radius: 0 0px 0 0;
+      }
+    }
+
+    ${isClassic(theme) || theme.name === THEMES.base ? applyClassicStyling : applyModernStyling}
+    ${selectableRowStyling}
+    ${selectedRowStyling}
+    ${highlightRowStyling}
+    ${dragRowStyling}
+  `;
+}
+
+function applyClassicStyling() {
+  return css`
+    border-color: #E7F1FC;
+    &:hover .common-input__input {
+      border-color: #E7F1FC;
+    }
+
+    &:nth-child(2n+1) {
+      ${StyledTableCell} {
+        background-color: #ffffff;
+      }
+    }
+
+    &:hover {
+      ${StyledTableCell} {
+        background-color: #E7F1FC;
+      }
+    }
+  `;
+}
+
+function applyModernStyling({ theme }) {
+  const { colors, table } = theme;
+  return css`   
+    ${StyledTableCell} {
+      padding-top: 0px;
+      padding-bottom: 0px;
+      background-color: ${colors.white};
+    }
+    &:hover {
+      ${StyledTableCell} {
+        background-color: ${table.primary};
+      }
+    }
+  `;
+}
+
+function selectableRowStyling() {
+  return css`
+    ${StyledTableCell}:first-child,
+    ${StyledTableHeader}:first-child {
+      &:not(.carbon-table-cell--select) {
+        padding-left: 15px;
+      }
+    }
+
+    .carbon-table-cell--select:first-child,
+    carbon-table-cell--select:first-child {
+        text-align: center;
+        width: 18px;
+      
+        .carbon-checkbox {
+          height: 15px;
+          padding-top: 0;
+        }
+      }
+    }
+  `;
+}
+
+function selectedRowStyling ({ theme }) {
+  const { table, colors, name } = theme;
+  return css`
+    .carbon-table-row--selected,
+    .carbon-table-row--selected:nth-child(odd),
+    .carbon-table-row--selected:hover {
+      ${StyledTableCell} {
+        background-color: ${isClassic(theme) || name === THEMES.base ? '#1573E6' : table.selected};
+        border-bottom-color: ${isClassic(theme) || name === THEMES.base ? '#255BC7' : table.selected};
+        color: ${isClassic(theme) || name === THEMES.base ? colors.white : ''};
+        position: relative;
+    
+        &:before {
+          background-color: ${isClassic(theme) || name === THEMES.base ? '#255BC7' : ''};
```

Comment:
could you override these styles in separate `classic` styles?

Reply:
I've separated some of it into classic and modern functions, I've left the odd line as just a simple ternary as they are isolated lines that will be easy to remove down the line